packages/ra-ui-materialui/src/input/FileInput.tsx

```diff
@@ -33,6 +33,7 @@ export const FileInput = (
         maxSize,
         minSize,
         multiple = false,
+        onSubmitRemove,
```

Comment:
I'd call it `validateFileRemoval`, and make it return either an empty promise (validated) or a rejected promise (failed).